rmw_fastrtps_cpp/src/functions.cpp

```diff
@@ -365,8 +400,14 @@ class topicnamesandtypesReaderListener : public ReaderListener
       tempMsg.length = change->serializedPayload.length;
       memcpy(tempMsg.buffer, change->serializedPayload.data, tempMsg.length);
       if (proxyData.readFromCDRMessage(&tempMsg)) {
+        auto partition_str = std::string("");
+        // don't use std::accumulate - schlemiel O(n2)
+        for (const auto & partition : proxyData.m_qos.m_partition.getNames()) {
```

Comment:
@Karsten1987 this should be removed/replace with checking for exactly one partition which contains the ros specific prefix, right?

I'll add a TODO while rebasing (https://github.com/ros2/rmw_fastrtps/pull/109 caused a conflict here)